build-on-all-workspaces.sh

```diff
@@ -0,0 +1,24 @@
+
+#!/bin/sh
+
+WORKSPACES="benches common protocols roles utils"
+
+for workspace in $WORKSPACES; do
+    echo "Executing build on: $workspace"
+    cargo build --manifest-path="$workspace/Cargo.toml" -- 
+    if [ $? -ne 0 ]; then
+        echo "Build found some errors in: $workspace"
+        exit 1
+    fi
+
+    echo "Running fmt on: $workspace"
+    (cd $workspace && cargo +nightly fmt)
+    if [ $? -ne 0 ]; then
+        echo "Fmt failed in: $workspace"
+        exit 1
+    fi
+done
+
+echo "build success, all tests passed!"
```

Comment:
the above script do not seem to execute any test, why you say all test passes?

Reply:
This is a mistake. This script is meant only for helping in an early stage of developing. I personally found it useful but I can remove it.

Reply:
updated by the way, now it prints just  `build success!"